### PR TITLE
Apply PHP 8.0 Syntax and constructor promotion

### PR DESCRIPTION
**Pull request by samsonasik**
Signed-off-by: Abdul Malik Ikhsan <samsonasik@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Since composer.json require php 8.0, php 8.0 syntax can constuctor promotion can be applied.

### CODE REVIEW

**Comment by samsonasik** [comment]
All green 🎉

File: src/Acl.php [hunk 1]
```diff
@@ -748,7 +748,7 @@ public function isAllowed($role = null, $resource = null, $privilege = null)
             // query on all privileges
             do {
                 // depth-first search on $role if it is not 'allRoles' pseudo-parent
-                if (null !== $role && null !== ($result = $this->roleDFSAllPrivileges($role, $resource, $privilege))) {
+                if (null !== $role && null !== ($result = $this->roleDFSAllPrivileges($role, $resource))) {
```
Comment (Ocramius) [inline src/Acl.php:751]: Something too much happened here

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Acl.php (hunk 1 above)
Comment (samsonasik) [reply src/Acl.php:751]: The `roleDFSAllPrivileges` params are only 2, and it passed 3 args on usage, so last unused arg passed can be removed

Comment (Ocramius) [reply src/Acl.php:751]: Eh, it's `protected`, so I'm not 100% sure about this 😬

I thin it's an acceptable change though: can't expect BC on this 👍

**Review by Ocramius — APPROVED** [review]
Thanks @samsonasik

**Pull request merged**